src/lib/plugins/live-cell/live-cell.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, Input, OnInit, SimpleChange, ChangeDetectionStrategy, NgZone } from '@angular/core';
+import { TemplateHostService } from '../../template/template-host.service';
+import { TdCoreDirective } from 'lib/main/core/body/td-core.directive';
+import { AppError } from '../../../core/infrastructure/error';
+
+@Component({
+	selector: 'q-grid-live-cell',
+	templateUrl: './live-cell.tpl.html',
+	providers: [TemplateHostService],
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LiveCellComponent implements OnInit {
+	@Input() cell: TdCoreDirective;
+	@Input() duration = 500;
+
+	changes: any;
+
+	constructor(private zone: NgZone) {
+	}
+
+	ngOnInit() {
+		if (!this.cell.changes) {
+			throw new AppError('q-grid-live-cell', 'Changes is not defined in live-cell.component.ts');
+		}
+
+		this.changes = this.getDifference(this.cell.changes);
```

Comment:
It should not be calculated here, just call it from template

Reply:
but i use this variable a lot of time in the template. Why should i calculate the value of a function many times, if I can calculate it once and put it in a variable?